src/libraries/Microsoft.PowerFx.Connectors/ArgumentMapper.cs

```diff
@@ -229,6 +229,28 @@ public ArgumentMapper(IEnumerable<OpenApiParameter> parameters, OpenApiOperation
             HiddenRequiredParamInfo = hiddenRequiredParams.ConvertAll(x => Convert(x, numberIsFloat: numberIsFloat)).Union(hiddenRequiredBodyParams.ConvertAll(x => Convert(x, numberIsFloat: numberIsFloat))).ToArray();
             OptionalParamInfo = optionalParams.ConvertAll(x => Convert(x, numberIsFloat: numberIsFloat)).Union(optionalBodyParams.ConvertAll(x => Convert(x, numberIsFloat: numberIsFloat))).ToArray();
 
+            // Validate we have no name conflict between required and optional parameters
+            // In case of conflict, we rename the optional parameter and add _1, _2, etc. until we have no conflict
+            List<string> requiredParamNames = RequiredParamInfo.Select(rpi => rpi.TypedName.Name.Value).ToList();
+            foreach (ServiceFunctionParameterTemplate opi in OptionalParamInfo)
```

Comment:
Is it possible for an optional param name to collide with a different optional param?

What about after rename, say we had:

Required: "Foo"
Optional: "Foo", "Foo_1"

Reply:
Notice that it's only between required parameters & optional *body* parameters (non-body parameters cannot use the same name as required parameters, per OpenAPI specification)
This is only due to the fact that we flatten body parameters for Power Apps (in a record, at the end...)

That said, we could construct such a strange API but I'm not considering it here.
I don't believe this is a valid scenario and while I appreciate your (very clever) comment, I don't want to complicate the code here - I'll add a comment though for this non-trivial possibility.

In parallel, and to mitigate this risk, I'm adding extra tests for tests all our connectors

Reply:
I think the real question here is: what does PowerApps do? 
We should match that behavior

Reply:
I'd need help to find where this is done. Would could help me here?

Reply:
Finally found where this logic is implemented in PA Client after a long research
This is in src\Cloud\DocumentServer.Core\Document\Importers\ServiceConfig\RestFunctionDefinitionBuilder.cs at line 1176 - CreateUniqueImpliedParameterName

They search for conflicts between any parameter (mandatory, optional) and any location (query, body) and then append "_n" where needed (starting with 1, and checking each number till no conflict)

I follow the same logic here, in a more restrictive manner
As the only example we need to support here is O365 Outlook connector and this code is sufficient, I think we're good enough for now and this code can be improved later when we'll have more testing around connectors.

Reply:
Added a comment in code